### PR TITLE
Fix cache usage for needles (again)

### PR DESCRIPTION
**Pull request by coolo**
Using regular expressions to replace paths is problematic - and using / is
even more problematic

But that's what happening if you trust reviewer's advise :)

### CODE REVIEW

**Comment by coolo** [comment]
This is now deployed on openqa.opensuse.org and works

**Comment by aaannz** [comment]
Whats wrong with `/`? This isn't bash, variable expansion is done after parsing substitute.

**Comment by coolo** [comment]
It confuses me :)

File: needle.pm [hunk 1]
```diff
@@ -202,7 +200,7 @@ sub wanted_ {
 }
 
 sub init {
-    my ($needledir, $shared_cache) = @_;
+    ($needledir, $shared_cache) = @_;
```
Comment (okurz) [inline needle.pm]: why this?

Comment (coolo) [inline needle.pm]: There was a misunderstanding: https://github.com/os-autoinst/os-autoinst/pull/638#discussion_r86559861

**Pull request merged**